packages/gatsby-plugin-mdx/src/compile-mdx.ts

```diff
@@ -1,27 +1,108 @@
+import type { NodePluginArgs } from "gatsby"
 import type { ProcessorOptions } from "@mdx-js/mdx"
-import type { IFileNode, IMdxNode } from "./types"
+import type { IFileNode, IMdxMetadata, IMdxNode } from "./types"
+
+import deepmerge from "deepmerge"
+
+import { enhanceMdxOptions, IMdxPluginOptions } from "./plugin-options"
 
 // Compiles MDX into JS
 // This could be replaced by @mdx-js/mdx if MDX compile would
 // accept custom data passed to the unified pipeline via processor.data()
-export default async function compileMDX(
-  source: string,
+export async function compileMDX(
   mdxNode: IMdxNode,
   fileNode: IFileNode,
   options: ProcessorOptions
-): Promise<string> {
-  const { createProcessor } = await import(`@mdx-js/mdx`)
-  const { VFile } = await import(`vfile`)
+): Promise<{ processedMDX: string; metadata: IMdxMetadata }> {
+  try {
+    const { createProcessor } = await import(`@mdx-js/mdx`)
+    const { VFile } = await import(`vfile`)
+
+    const processor = createProcessor(options)
+
+    // If we could pass this via MDX loader config, this whole custom loader might be obsolete.
```

Comment:
Can we link an issue/discussion here?

Reply:
Unfortunately, this comment is a little bit out of date. There is another difference to their loader:

https://github.com/gatsbyjs/gatsby/pull/35873/files/d673d5c5853c2993c0ee9abdb36da9d66c23f31c#diff-fbbdffa47e09f9a24b72a54d1c082d613787d81ae0666e5c3db2f9cc04581b6aR30

We would need to get that path in as well.

Reply:
I rephrased it